SEBrowser/Controllers/OpenXDAController.cs

```diff
@@ -219,32 +225,47 @@ private string getEventTypeFilter(EventSearchPostData postData)
 
         private string getPhaseFilter(EventSearchPostData postData)
         {
-            List<string> phase = new List<string>();
-
-            if (postData.PhaseA && postData.PhaseB && postData.PhaseC)
-                return "";
-            if (!postData.PhaseA && !postData.PhaseB && !postData.PhaseC)
+            Dictionary<string, bool> phases = new Dictionary<string, bool> 
+            {
+                ["AN"] = postData.PhaseAN,
+                ["BN"] = postData.PhaseBN,
+                ["CN"] = postData.PhaseCN,
+                ["AB"] = postData.PhaseAB,
+                ["BC"] = postData.PhaseBC,
+                ["ABG"] = postData.PhaseABG,
+                ["BCG"] = postData.PhaseBCG,
+                ["ABC"] = postData.PhaseABC,
+                ["ABCG"] = postData.PhaseABCG
+            };
+
+            bool allTrue = true;
+            bool allFalse = true;
+            foreach (var value in phases.Values) 

```

Comment:
You can use LINQ and do something like `phases.Select(item => item.Value).Some()` to determine if all are `true` or  `false`. you may need to think through the exact statement to get `allTrue` and `allFalse` but it should be a oneliner